### ISSUE TITLE
XmlElement is not honored

### ISSUE THREAD

**Issue by cubidobusinesssolutions** [DigDes/SoapCore#243]
`XmlElementAttribute` is not working on array parameters

Code:
```cs
[ServiceContract(Namespace ="urn:test:1")]
public interface ITestService
{
    [OperationContract]
    TestResult ExecuteFunction([XmlElement("parameterArray")]string[] parameterArray);
}
```

This will parse an empty array.

**Comment by cubidobusinesssolutions** [comment]
I will provide a PR for this and #180 in a minute

**Comment by kotovaleksandr** [comment]
I trying to understand how XmlElement work and observe what array parameters with XmlSerializer never worked. 
In #248 user observe changed behavior of SoapCore, but i don't know how it works previously.   
I'm tried to modify your code for parsing array parameters and write some tests (look in https://github.com/DigDes/SoapCore/commit/cca43e19f9e166d8c5d7f08d4a0b964d497264c1). For case where array parameters used without any attributes tests passed and for first view works fine (exists problem with system element types such as string, int - which serialized in lower case tags, but in code types named with first upper case letter, fixed with check for System namespace).

But, i added to test service methods:
```csharp
[OperationContract]
[XmlSerializerFormat]
string[] PingStringArrayWithXmlArray([XmlElement("arrayItem")]string[] array);

[OperationContract]
[XmlSerializerFormat]
ComplexModel1[] PingComplexModelArrayWithXmlArray([XmlArrayItem("arr1")]ComplexModel1[] models, [XmlElement("arr2")]ComplexModel2[] models2);
``` 
And tests generates identical requests to server in both cases: with XmlArray (XmlElement) attribute and without them. Your code wait request body with different xml and test failed. Whats client you used for make requests to SoapCore?

If you can look to code and fix tests and SoapCore code (or write new correct tests) - i would thank you very much. I seem to be at a dead end.

**Comment by cubidobusinesssolutions** [comment]
@kotovaleksandr I think this can be referenced (and closed) in latest [Release 0.9.9.5](https://github.com/DigDes/SoapCore/releases/tag/0.9.9.5), right?

**Comment by kotovaleksandr** [comment]
No, this issue founded in 0.9.9.5 release, see original issue #248. Your PR breaks serialization of array responses.

**Comment by cubidobusinesssolutions** [comment]
@kotovaleksandr Sry, I didn't see your comment until now
> Whats client you used for make requests to SoapCore?
Visual Studio Web Reference from .wsdl file, see https://docs.microsoft.com/en-us/sharepoint/dev/general-development/step-2-adding-a-web-reference

**Comment by MCFHTAGENTS** [comment]
Hi, I see a new release has just been done (0.0.9.9.6) - that doesn't seem to address this. Anyway we can get this through?

**Comment by gorillapower** [comment]
We are also experiencing this issue

**Comment by cubidobusinesssolutions** [comment]
@kotovaleksandr Any updates on this? I am still locked on version 0.9.9.5, because in newer versions, string arrays are not deserialized properly for legacy clients.

**Comment by kotovaleksandr** [comment]
> @kotovaleksandr Any updates on this? I am still locked on version 0.9.9.5, because in newer versions, string arrays are not deserialized properly for legacy clients.

No changes:(

**Comment by vinydanylo** [comment]
Just a remark, when you try to change the element name, all the properties from ChildClass will have the same name as its ElementName on the parentClass, even though it preserves their type.
```c#
[XmlType("parentclass")]
[Serializable]
public class ParentClass
{
    [XmlIgnore]
    public int Id { get; set; }

    [XmlElement(ElementName="myclasselement")]
    public ChildClass MyClassElement { get; set; }
}

[XmlType("childclass")]
[Serializable]
public class ChildClass
{
    [XmlElement(ElementName="mydate")]
    public DateTime MyDate { get; set; }

    [XmlElement(ElementName="myint")]
    public int MyInt { get; set; }
}
```

```xml
<xsd:complexType name="parentclass">
    <xsd:sequence>
        <xsd:element minOccurs="0" maxOccurs="1" name="InvoiceDetail" type="tns:childclass"/>
    </xsd:sequence>
</xsd:complexType>

<xsd:complexType name="childclass">
    <xsd:sequence>
        <xsd:element minOccurs="1" maxOccurs="1" name="myclasselement" type="xsd:dateTime"/>
        <xsd:element minOccurs="1" maxOccurs="1" name="myclasselement" type="xsd:int"/>
    </xsd:sequence>
</xsd:complexType>
```

I hope I could explain well =)

**Comment by cubidobusinesssolutions** [comment]
TL;DR Instead of applying `XmlElementAttribute` directly to parameters, wrap parameters in request model class:
```CSharp
public interface IService
{
  response action(actionRequest request);
}
[MessageContract]
public class actionRequest
{
  [MessageBodyMember]
  [XmlElement]
  public string param1;
}
```

Finally, I upgraded to the latest version with only minor issues (see #591). After some digging, I stumbled accross the following comment in Connected Service generated code (namespaces omitted for brevity)

```CSharp
[GeneratedCodeAttribute("Microsoft.Tools.ServiceModel.Svcutil", "2.0.2")]
public interface MyService
{
  // CODEGEN: Parameter 'param1' requires additional schema information that cannot be captured using the parameter mode. The specific attribute is 'Microsoft.Xml.Serialization.XmlElementAttribute'.
  [OperationContractAttribute(Action="", ReplyAction="*")]
  method1Response method1(method1 request);
}
[DebuggerStepThroughAttribute()]
[GeneratedCodeAttribute("Microsoft.Tools.ServiceModel.Svcutil", "2.0.2")]
[EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Advanced)]
[MessageContractAttribute(WrapperName="method1", IsWrapped=true)]
public partial class method1
{
  [MessageBodyMemberAttribute(Order=0)]
  [XmlElementAttribute(Form=System.Xml.Schema.XmlSchemaForm.Unqualified)]
  public string param1;
}
```

`XmlElementAttribute` was never designed to do what I asked for^^
